pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller_test.go

```diff
@@ -0,0 +1,242 @@
+			}),
+		Entry("events with nil node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: nil,
+				Reconcile:    true,
+			}),
+		Entry("events with empty node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{},
+				Reconcile:    true,
+			}),
+		Entry("events with matching node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: true,
+			}),
+		Entry("events with missing label at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("events with different label value at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar1",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("node not found",
+			predicateCase{
+				ObjNodeName:  "node02",
+				NodeLabels:   map[string]string{},
+				NodeSelector: map[string]string{},
+				Reconcile:    false,
+			}),
+	)
+})
+
+var _ = Describe("NodeNetworkConfigurationPolicy controller reconciler", func() {
+	var (
+		name          = "nodenetworkconfigurationpolicy-test"
+		namespace     = "default"
+		numberOfNodes = 2
+		nodes         []nmstatev1alpha1.NodeNetworkState
+
+		cl      client.Client
+		r       *ReconcileNodeNetworkConfigurationPolicy
+		req     reconcile.Request
+		policy1 = nmstatev1alpha1.NodeNetworkConfigurationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+		res reconcile.Result
+	)
+
+	BeforeEach(func() {
+
+		By("populate the NodeNetworkState for nodes")
+		nodes = nil
+		for n := 1; n <= numberOfNodes; n++ {
+			nodes = append(nodes, nmstatev1alpha1.NodeNetworkState{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("node%02d", n),
+					Namespace: namespace,
+				},
+			})
+		}
+
+		// Mock request to simulate Reconcile() being called on an event for a
+		// watched resource .
+		req = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+
+		// Register operator types with the runtime scheme.
+		By("register state and policies types")
+		s := scheme.Scheme
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion, &policy1)
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion, &nodes[0])
```

Comment:
you should just initialize and pass the struct here. do not assume the list has expexted type and is not empty

Reply:
Done